### PR TITLE
groups factories options and move *areEqual defaults into connect

### PR DESCRIPTION
**Pull request by jimbolla**
take 2

### CODE REVIEW

**Comment by jimbolla** [comment]
I'm confused by the code coverage fail here, since it doesn't seem to line up with what I changed.

File: src/connect/connect.js [hunk 1]
```diff
@@ -29,16 +30,24 @@ function match(arg, factories) {
...
-    mapStateToPropsFactories = defaultMapStateToPropsFactories,
-    mapDispatchToPropsFactories = defaultMapDispatchToPropsFactories,
-    mergePropsFactories = defaultMergePropsFactories,
-    selectorFactory = defaultSelectorFactory,
     pure = true,
+    areStatesEqual = strictEqual,
+    areOwnPropsEqual = shallowEqual,
+    areStatePropsEqual = shallowEqual,
+    areMergedPropsEqual = shallowEqual,
+    factories: {
```
Comment (timdorr) [inline src/connect/connect.js]: I don't really agree with this. I'd rather keep the options map flat, even though it's somewhat large. It keeps things relatively symmetrical when dealing with them.

Comment (jimbolla) [inline src/connect/connect.js]: As I'm writing the docs for the new connect features, i'm realizing the 4 factories are different than the other options; they don't make as much sense to use at the individual call sites like the other options. They're intention is to be used in a custom wrapper function like 'customConnect`. That's why I'm trying to separate them from the other options. Maybe they shouldn't be in options at all. Maybe I should use a customization pattern like [redux-thunk](https://github.com/gaearon/redux-thunk/blob/master/src/index.js) or [reselect's createSelector](https://github.com/reactjs/reselect/blob/master/src/index.js#L66). Any thoughts on that?

**Comment by jimbolla** [comment]
Ok. I'm happier with this. Now the "factories" aren't part of the main API; they're only accessible via `createConnect` which isn't exported from `index.js`. I'd like to get some feedback on how other devs might use it before making it official. We can expose it later in a minor release. The upside to this is I can defer writing up docs for it until later.

**Comment by timdorr** [comment]
Putting this one in too.

**Pull request merged**